### PR TITLE
Add a loop and retry for the email modal

### PR DESCRIPTION
**Pull request by pattonwebz**
This prevents a JS error and will fail gracefully rather than producing any error a user would see or have issues because of it erroring.

Fixes: #721 

## Checklist

- [x] PR is linked to the main issue in the repo
- [ ] Tests are added that cover changes

### CODE REVIEW

File: src/emailOptIn/modal.js [hunk 1]
```diff
@@ -15,28 +15,39 @@ export const initOptInModal = () => {
...
 };
 
+const bindFocusTrap = () => {
+	const modal = document.getElementById( 'TB_window' );
+	const closeicon = modal?.querySelector( '.tb-close-icon' );
+	if ( ! modal || ! close ) {
+		return false;
+	}
+
+	closeicon.setAttribute( 'aria-hidden', 'true' );
```
Comment (SteveJonesDev) [inline src/emailOptIn/modal.js:36]: Corrected `close icon` to `closeIcon` for consistency

```suggestion
	closeIcon.setAttribute( 'aria-hidden', 'true' );
```

File: src/emailOptIn/modal.js [hunk 2]
```diff
@@ -15,28 +15,39 @@ export const initOptInModal = () => {
...
+				clearInterval( intervalId );
+			}
+			attempts++;
+		}, 250 );
 	};
 };
 
+const bindFocusTrap = () => {
+	const modal = document.getElementById( 'TB_window' );
+	const closeicon = modal?.querySelector( '.tb-close-icon' );
```
Comment (SteveJonesDev) [inline src/emailOptIn/modal.js:31]: Corrected `close` icon to `closeIcon` for consistency.

```suggestion
	const closeIcon = modal?.querySelector( '.tb-close-icon' );
```

File: src/emailOptIn/modal.js [hunk 3]
```diff
@@ -15,28 +15,39 @@ export const initOptInModal = () => {
...
+	if ( ! modal || ! close ) {
```
Comment (SteveJonesDev) [inline src/emailOptIn/modal.js:32]: Corrected Null checks.

```suggestion
	if ( ! modal || ! closeicon ) {
```

**Review by SteveJonesDev — CHANGES_REQUESTED** [review]

File: src/emailOptIn/modal.js [hunk 4]
```diff
@@ -15,28 +15,39 @@ export const initOptInModal = () => {
...
+			}
+			attempts++;
+		}, 250 );
 	};
 };
 
+const bindFocusTrap = () => {
+	const modal = document.getElementById( 'TB_window' );
+	const closeIcon = modal?.querySelector( '.tb-close-icon' );
+	if ( ! modal || ! close ) {
```
Comment (pattonwebz) [inline src/emailOptIn/modal.js:32]:
```suggestion
	const closeIcon = modal?.querySelector( '.tb-close-icon' );
	if ( ! modal || ! closeIcon ) {
```

File: src/emailOptIn/modal.js (hunk 3 above)
Comment (pattonwebz) [reply src/emailOptIn/modal.js]: Fixed this in [`19035eb` (#722)](https://github.com/equalizedigital/accessibility-checker/pull/722/commits/19035eb0cd93ff6d7a7786823003bbee3f7fc1b7) and used camellCase for it

**Comment by pattonwebz** [comment]
@SteveJonesDev can you take another look now with the changes suggested commited?

**Review by SteveJonesDev — APPROVED** [review]

**Pull request merged**